crates/uv-resolver/src/lock/export/cyclonedx_json.rs

```diff
@@ -0,0 +1,418 @@
+                PackageType::Workspace(path)
+            } else {
+                PackageType::Dependency
+            };
+            component_builder.create_component(node.package, package_type, Some(&node.marker))
+        })
+        .collect::<Vec<_>>();
+
+    let mut dependencies = create_dependencies(&nodes, &component_builder);
+
+    // With `--all-packages`, use synthetic root which depends on workspace root and all workspace members.
+    // This ensures that we don't have any dangling components resulting from workspace packages not depended on by the workspace root.
+    if all_packages {
+        let synthetic_root = component_builder.create_synthetic_root_component(root);
+        let synthetic_root_bom_ref = synthetic_root
+            .bom_ref
+            .clone()
+            .expect("bom-ref should always exist");
+        let workspace_root = metadata.component.replace(synthetic_root);
+
+        if let Some(workspace_root) = workspace_root {
+            components.push(workspace_root);
+        }
+
+        dependencies.push(Dependency {
+            dependency_ref: synthetic_root_bom_ref,
+            dependencies: workspace_member_ids
+                .iter()
+                .filter_map(|c| component_builder.get_component(c))
+                .map(|c| c.bom_ref.clone().expect("bom-ref should always exist"))
+                .collect(),
+        });
+    }
+
+    let bom = Bom {
+        metadata: Some(metadata),
+        components: Some(Components(components)),
+        dependencies: Some(Dependencies(dependencies)),
+        ..Bom::default()
+    };
+
+    Ok(bom)
+}
+
+fn create_dependencies(
+    nodes: &[ExportableRequirement<'_>],
+    component_builder: &ComponentBuilder,
+) -> Vec<Dependency> {
+    nodes
+        .iter()
+        .map(|node| {
+            let component = component_builder
+                .get_component(&node.package.id)
+                .expect("All nodes should have been added to map");
+
+            let immediate_deps = &node.package.dependencies;
+            let optional_deps = node.package.optional_dependencies.values().flatten();
+            let dep_groups = node.package.dependency_groups.values().flatten();
+
+            let package_deps = immediate_deps
+                .iter()
+                .chain(optional_deps)
+                .chain(dep_groups)
+                .filter_map(|dep| component_builder.get_component(&dep.package_id));
+
+            let bom_refs = package_deps
+                .map(|p| p.bom_ref.clone().expect("bom-ref should always exist"))
+                .sorted_unstable()
+                .unique()
+                .collect();
+
+            Dependency {
+                dependency_ref: component
+                    .bom_ref
+                    .clone()
+                    .expect("bom-ref should always exist"),
+                dependencies: bom_refs,
+            }
+        })
+        .collect()
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
```

Comment:
Per above, this should work 🙂 

```suggestion
#[derive(Copy, Clone, Debug, Eq, PartialEq)]
```